pyface/testing/gui_test_case.py

```diff
@@ -0,0 +1,443 @@
+
+    def event_loop_with_timeout(self, repeat=1, timeout=10):
+        """ Run the event loop for timeout seconds.
+        """
+        self.event_loop_helper.event_loop_with_timeout(repeat, timeout)
+
+    def destroy_control(self, control, timeout=1.0):
+        """ Schedule a toolkit control for destruction and run the event loop.
+
+        Parameters
+        ----------
+        control : toolkit control
+            The control to destroy.
+        timeout : float
+            The number of seconds to run the event loop in the event that the
+            control is not destroyed.
+
+        Raises
+        ------
+        ConditionTimeoutError
+            If the timeout occurs before the widget is destroyed.
+        """
+        destroy_later(control)
+        self.event_loop_until_control_destroyed(control, timeout)
+
+    def destroy_widget(self, widget, timeout=1.0):
+        """ Schedule a Widget for destruction and run the event loop.
+
+        Parameters
+        ----------
+        control : IWidget
+            The widget to destroy.
+        timeout : float
+            The number of seconds to run the event loop in the event that the
+            widget is not destroyed.
+
+        Raises
+        ------
+        ConditionTimeoutError
+            If the timeout occurs before the widget is destroyed.
+        """
+        if widget.control is not None:
+            control = widget.control
+            widget.destroy()
+            self.event_loop_until_control_destroyed(control, timeout)
+
+    # ------------------------------------------------------------------------
+    #  'TestCase' protocol
+    # ------------------------------------------------------------------------
+
+    def setUp(self):
+        """ Setup the test case for GUI interactions.
+        """
+        self.gui = GUI()
+        self.app = self.gui.app
+        self.event_loop_helper = EventLoopHelper(gui=self.gui)
+
+        if not os.environ.get('PYFACE_PATCH_ACTIVATE', False):
+            original_activated = Window.activate
+
+            def activate_patch(self, should_raise=True):
+                return original_activated(False)
+
+            self._raise_patch = mock.patch.object(
+                Window, 'activate', activate_patch)
+            self._raise_patch.start()
+        else:
+            self._raise_patch = None
+
+        self.gui.quit_on_last_window_close = False
+
+        # clean-up actions (LIFO)
+        self.addCleanup(self._delete_attrs, "gui", "app", "event_loop_helper")
+        self.addCleanup(self._restore_window_activate)
+        self.addCleanup(self._restore_quit_on_last_window_close)
+        self.addCleanup(self.gui.clear_event_queue)
+        self.addCleanup(self.gui.process_events)
+        self.addCleanup(self._close_top_level_windows)
+        self.addCleanup(self._gc_collect)
+        self.addCleanup(self.event_loop_helper.event_loop, 5, False)
+
+        super(GuiTestCase, self).setUp()
```

Comment:
> But that likely won't work

Okay, thinking about this, I'm not sure any more. It may work after all. The `UnittestTools` class will just be effectively invisible to the `super` machinery, because it doesn't implement `setUp` and `tearDown` at all.

Can we have a test for multiple inheritance cases that verifies that both the parent class `setUp` methods get called as expected? One way that this could potentially break is if some future version of `UnittestTools` *does* decide to implement `setUp`, so it would be useful to have a test that alerts us to the breakage.